src/components/Card/Card.test.tsx

```diff
@@ -1,37 +1,98 @@
-import { render } from '@testing-library/react';
 import React from 'react';
+import { render, screen } from '@testing-library/react';
 
 import { Card, CardProps } from './';
 
+import styles from './Card.module.scss';
+
 const DEFAULT_PROPS: CardProps = {
-  title: 'Title Text',
-  value: '123',
-  bottomText: 'Bottom Text'
+  title: '',
+  value: '',
+  bottomText: undefined
 };
 
+const MOCK_SKELETON_ID = 'mock-skeleton';
+
 jest.mock('../Skeleton', () => ({
-  Skeleton: () => <span>Skeleton</span>
+  Skeleton: () => <div data-testid={MOCK_SKELETON_ID} />
 }));
 
-const renderComponent = (props?: CardProps) => render(<Card {...props} />);
+describe('<Card />', () => {
+  test('should only render text in the correct order - title, value, bottom text', () => {
+    const { container } = render(<Card {...DEFAULT_PROPS} title={'1,'} value={'2,'} bottomText={'3'} />);
+    expect(container).toHaveTextContent('1,2,3');
+  });
 
-test('should work with synchronous data', () => {
-  const { getByText } = renderComponent(DEFAULT_PROPS);
-  expect(getByText('Title Text')).toBeInTheDocument();
-  expect(getByText('123')).toBeInTheDocument();
-  expect(getByText('Bottom Text')).toBeInTheDocument();
-});
+  describe('semantic tags', () => {
+    test('should render title as <label>', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} title={'mock title'} />);
+      const label = container.getElementsByTagName('label');
+      expect(label.length).toBe(1);
+      expect(label[0]).toHaveTextContent('mock title');
+    });
+  });
 
-test('should show skeleton when loading asynchronous data', () => {
-  const { getByText } = renderComponent({ ...DEFAULT_PROPS, value: { isLoading: true } });
-  expect(getByText('Skeleton')).toBeInTheDocument();
-});
+  describe('class names', () => {
+    test('should apply Title class to only the title element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} title={'mock title'} />);
+      const title = container.getElementsByClassName(styles.Title);
+      expect(title.length).toBe(1);
+      expect(title[0]).toHaveTextContent('mock title');
+    });
+
+    test('should only apply Value class to only the value element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} value={'mock value'} />);
+      const value = container.getElementsByClassName(styles.Value);
+      expect(value.length).toBe(1);
+      expect(value[0]).toHaveTextContent('mock value');
+    });
+
+    test('should only apply BottomText class to the bottom text element', () => {
+      const { container } = render(<Card {...DEFAULT_PROPS} bottomText={'mock bottom text'} />);
+      const bottomText = container.getElementsByClassName(styles.BottomText);
+      expect(bottomText.length).toBe(1);
+      expect(bottomText[0]).toHaveTextContent('mock bottom text');
```

Comment:
As we're only testing ```BottomText``` class is applied, would it make more sense to reduce this test to something like:

```
test('should only apply BottomText class to the bottom text element', () => {
  const { container } = render(<Card {...DEFAULT_PROPS} bottomText={'mock bottom text'} />);
  expect(container.getElementsByClassName(styles.BottomText)).toBeTruthy();
});
```

I'd say we could separate the expects for elements length and text content in this and the above tests if our aim is to ensure each test is scoped to exactly what it asks.

Reply:
It's tricky - the suggested checks that the class was applied to an element, but not the correct element. I found this a pretty tough thing to test without using `testId` - perhaps this is a situation where `testId` works well!

e.g.
```
render(...);
const bottomText = screen.getByTestId(...);
expect(bottomText).toHaveClass(...);
```

Reply:
Ahhh no, I see what you mean. Checking the length covers the 'only' part of the scope. Probably don't need to check the text content, unless that hasn't been covered in another test, in that case I'd leave this as is 😁